guides/building-a-middleware-from-scratch.md

```diff
@@ -519,6 +519,94 @@ where
 }
 ```
 
+Create a layer for usage:
+```rs
+impl TimeoutLayer {
+    /// Create a timeout from a duration
+    pub fn new(timeout: Duration) -> Self {
+        TimeoutLayer { timeout }
+    }
+}
+
+impl<S> Layer<S> for TimeoutLayer {
+    type Service = Timeout<S>;
+
+    fn layer(&self, service: S) -> Self::Service {
+        Timeout::new(service, self.timeout)
+    }
+}
+```
+
+Usage
+```rs
+use axum::{
+    Router,
+    routing::get,
+};
+use std::time::Duration;
+use tower::ServiceBuilder;
+
+let app = Router::new()
+    .route("/", get(|| async {}))
+    .layer(
+        ServiceBuilder::new()
+            .layer(TimeoutLayer::new(Duration::from_secs(5)))
+    );
+```
+
+But wait, we get a compile error:
+```
+error[E0277]: the trait bound `Infallible: From<Box<(dyn std::error::Error + Send + Sync + 'static)>>` is not satisfied
+   --> src/lib.rs:123:25
+    |
+123 |     Router::new().layer(ServiceBuilder::new().layer(TimeoutLayer::new(Duration::from_secs(5))))
+    |                   ----- ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `From<Box<(dyn std::error::Error + Send + Sync + 'static)>>` is not implemented for `Infallible`
+    |                   |
+    |                   required by a bound introduced by this call
+    |
+```
+
+That's because there's an attempt to convert a Box<dyn StdError + Send + Sync + 'static> into Infallible. The Infallible type is used to signify that a function will never return an error.
+> Axum’s error handling model requires handlers to always return a response. However middleware is one possible way to introduce errors into an application. If hyper receives an error the connection will be closed without sending a response. Thus axum requires those errors to be handled gracefully. - [doc](https://docs.rs/axum/latest/axum/middleware/index.html#error-handling-for-middleware)
```

Comment:
Should we really add specific framework mentions to these fundamental docs? Is that done elsewhere? Just asking, as I'm not sure of the value of it, given not everyone might be familiar with hyper and/or axum.

Reply:
@GlenDC That's a valid point. Apologies, I'm kinda new to the tokio ecosystem and didn't fully realize that tower is incredibly framework agnostic. The motivation was to provide an example of usage and what issues might come up when integrating with Axum. But you're right in that:
> given not everyone might be familiar with hyper and/or axum.

Maybe this can find a home in one of Axum's examples

I can close out this PR and the accompanying Issue